pkg/occlient/occlient_test.go

```diff
@@ -5007,6 +5007,116 @@ func TestWaitAndGetDC(t *testing.T) {
 	}
 }
 
+func TestCreateBuildConfigWithBinaryInput(t *testing.T) {
+	type args struct {
+		commonObjectMeta metav1.ObjectMeta
+		kind             string
+		namespace        string
+		name             string
+		scripts          string
+		outputimageTag   string
+		incremental      bool
+		envVars          []corev1.EnvVar
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		actions int
+	}{
+		{
+			name: "Case 1 - Generate and create the BuildConfig",
+			args: args{
+				kind:           "ImageStreamTag",
+				namespace:      "openshift",
+				name:           "ruby:latest",
+				outputimageTag: "nodeimage:latest",
+				scripts:        "",
+				incremental:    false,
+				commonObjectMeta: metav1.ObjectMeta{
+					Name: "ruby",
+				},
+				envVars: []corev1.EnvVar{
+					{
+						Name:  "key",
+						Value: "value",
+					},
+					{
+						Name:  "key1",
+						Value: "value1",
+					},
+				},
+			},
+			wantErr: false,
+			actions: 4,
+		},
+		{
+			name: "Case 2 - Generate and create the BuildConfig but fail with unable to find image name",
+			args: args{
+				kind:           "ImageStreamTag",
+				namespace:      "testing",
+				name:           "fakeimagename:notlatest",
+				outputimageTag: "nodeimage:latest",
+				scripts:        "",
+				incremental:    false,
+				commonObjectMeta: metav1.ObjectMeta{
+					Name: "ruby",
+				},
+				envVars: []corev1.EnvVar{
+					{
+						Name:  "key",
+						Value: "value",
+					},
+					{
+						Name:  "key1",
+						Value: "value1",
+					},
+				},
+			},
+			wantErr: true,
+			actions: 4,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, fakeClientSet := FakeNew()
+
+			fakeClientSet.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStream(tt.args.commonObjectMeta.Name, tt.args.commonObjectMeta.Namespace, []string{"latest"}), nil
+			})
+			// Run function CreateBuildConfig
+			bc, err := fakeClient.CreateSourceBuildConfigWithBinaryInput(tt.args.commonObjectMeta, tt.args.kind, tt.args.namespace, tt.args.name, tt.args.outputimageTag, tt.args.scripts, tt.args.incremental, []corev1.EnvVar{}, "DockerImage", "")
+			if err == nil && !tt.wantErr {
+				// Check to see how many actions are being ran
+				if (len(fakeClientSet.ImageClientset.Actions()) != tt.actions) && !tt.wantErr {
+					t.Errorf("expected %v action(s) in CreateSourceBuildConfigWithBinaryInput got %v: %v", tt.actions, len(fakeClientSet.ImageClientset.Actions()), fakeClientSet.ImageClientset.Actions())
+				}
+
+				// Check to see that names match
+				if bc.ObjectMeta.Name != tt.args.commonObjectMeta.Name {
+					t.Errorf("Expected buildConfig name %s, got '%s'", tt.args.commonObjectMeta.Name, bc.ObjectMeta.Name)
+				}
```

Comment:
This should be checked by doing ```Get``` from the prepend reactor instead of just comparing returned object from the api call